### PR TITLE
feat(openapi): implement "open_api_do" for the new version of OpenAPI

### PR DESCRIPTION
**Pull request by willandfree**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2j40ybw OpenAPIv2 对接 —— Dataset](https://app.clickup.com/t/2j40ybw)

File: graviti/openapi/request.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from typing import Any
+from uuid import uuid4
+
+from requests.models import Response
+from tensorbay.exception import ResponseError, ResponseErrorDistributor
+from tensorbay.utility import config, get_session
+
+from graviti.__version__ import __version__
+
+URL_PATH_PREFIX = "gatewayv2/graviti-open-api/v2gi"
```
Comment (graczhual) [inline graviti/openapi/request.py:17]: api/v1

File: graviti/openapi/request.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
```
Comment (linjiX) [inline graviti/openapi/request.py:1]: we need a more accurate commit message
ask @graczhual for help.

File: graviti/openapi/request.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+from typing import Any
+from uuid import uuid4
+
+from requests.models import Response
+from tensorbay.exception import ResponseError, ResponseErrorDistributor
+from tensorbay.utility import config, get_session
+
+from graviti.__version__ import __version__
+
+URL_PATH_PREFIX = "gatewayv2/graviti-open-api/v1"
```
Comment (graczhual) [inline graviti/openapi/request.py:17]: do not change?

File: graviti/openapi/request.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
```
Comment (graczhual) [inline graviti/openapi/request.py:3]: add __init__.py

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/__init__.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Graviti Python SDK."""
```
Comment (graczhual) [inline graviti/openapi/__init__.py:6]: OpenAPI module

File: graviti/openapi/__init__.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+from graviti.__version__ import __version__
+
+__all__ = ["__version__"]
```
Comment (graczhual) [inline graviti/openapi/__init__.py:10]: do not need this

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/__init__.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
```
Comment (graczhual) [inline graviti/openapi/__init__.py:4]: add one blank line

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/request.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+from requests.models import Response
+from tensorbay.exception import ResponseError, ResponseErrorDistributor
+from tensorbay.utility import config, get_session
+
+from graviti.__version__ import __version__
+
+URL_PATH_PREFIX = "api/v1"
+
+
+def open_api_do(url: str, access_key: str, method: str, **kwargs: Any) -> Response:
```
Comment (graczhual) [inline graviti/openapi/request.py:20]: The order is inconsistent with arguments.
The order should be: access_key, url, method, **kwargs

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/request.py (hunk 8 above)
Comment (linjiX) [reply graviti/openapi/request.py:20]: I think `method` should be the first?

Comment (graczhual) [reply graviti/openapi/request.py:20]: Yes, then the order should be method, access_key, url, **kwargs

File: graviti/openapi/request.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+
+def open_api_do(method: str, access_key: str, url: str, **kwargs: Any) -> Response:
+    """Send a request to the Graviti OpenAPI.
+
+    Arguments:
+        access_key: User's access key.
+        method: The method of the request.
+        url: The URL of the graviti website.
+        **kwargs: Extra keyword arguments to send in the POST request.
```
Comment (graczhual) [inline graviti/openapi/request.py:27]: change the order

**Review by graczhual — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**